### ISSUE TITLE
Errors when computing partial charges for PDB files

### ISSUE THREAD

**Issue by taimir** [rdkit/rdkit#1253]
Hi there,

we want to compute partial charges for proteins in PDB format. We use the [Gasteiger](http://www.rdkit.org/Python_Docs/rdkit.Chem.rdPartialCharges-module.html#ComputeGasteigerCharges) algorithm provided in  `rdkit.Chem.rdPartialCharges`, but *quite a lot of* files cause errors. Maybe you could recommend another implementation or another algorithm to use?

Here is an example error for the PDB entry `1a0a`:

```
ERROR: No Gasteiger Partial Charge parameters for Element: P Mode:
```

After some investigation, we've tracked the problem down to the following: there are hard-coded partial charges that are used as initial parameterization of the Gasteiger algorithm ([see defaultParamData here](https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/PartialCharges/GasteigerParams.cpp)). The problem is that the hard-coded partial charges are associated with a pair of the form `(atom_name, mode)` where mode seems to be the `Hybridization` of the atom. In the file `1a0a`, the problem is that the P atom in question has hybridization SP3D, while in the hardcoded parameters there is a partial charge only for the pair (P, SP3), but not for (P, SP3D). And so the `Gasteiger` algorithm fails with the error above.

Any ideas of how we could work around this (or tips on other ways to compute the charges) would be appreciated, thanks!

**Comment by levipierce** [comment]
Taimir were you able to come up with any solution?  I am also seeing this problem

**Comment by taimir** [comment]
Unfortunately no, I had to resort to just ignoring the problematic PDB files for the time being. I do not think there's any proper way to solve this in `rdkit`, other than changing the `rdkit` code that causes the issue (what I referenced above). 

Possible solutions could either modify the PDB files in some reliable way so that it all works out, or suggest an alternative algorithm for the partial charge computation - so far I haven't come across either one.

**Comment by greglandrum** [comment]
Yeah, I'm afraid that there's not a lot we can do within the constraints of the algorithm that's currently there. Handling different element/hybridization types would require either a new partial charge calculation scheme or a reparameterization of the existing scheme (which is likely almost as much work).
